src/spatialdata/_core/operations/vectorize.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+from functools import singledispatch
+
+import numpy as np
+import pandas as pd
+from geopandas import GeoDataFrame
+from multiscale_spatial_image import MultiscaleSpatialImage
+from shapely import MultiPolygon, Point, Polygon
+from spatial_image import SpatialImage
+
+from spatialdata._core.centroids import get_centroids
+from spatialdata._core.operations.aggregate import aggregate
+from spatialdata.models import (
+    Image2DModel,
+    Image3DModel,
+    Labels3DModel,
+    ShapesModel,
+    SpatialElement,
+    get_axes_names,
+    get_model,
+)
+from spatialdata.transformations.operations import get_transformation
+from spatialdata.transformations.transformations import Identity
+
+INTRINSIC_COORDINATE_SYSTEM = "__intrinsic"
+
+
+@singledispatch
+def to_circles(
+    data: SpatialElement,
+) -> GeoDataFrame:
+    """
+    Convert a set of geometries (2D/3D labels, 2D shapes) to approximated circles/spheres.
+
+    Parameters
+    ----------
+    data
+        The SpatialElement representing the geometries to approximate as circles/spheres.
+
+    Returns
+    -------
+    The approximated circles/spheres.
+
+    Notes
+    -----
+    The approximation is done by computing the centroids and the area/volume of the geometries. The geometries are then
+    replaced by circles/spheres with the same centroids and area/volume.
+    """
+    raise RuntimeError(f"Unsupported type: {type(data)}")
+
+
+@to_circles.register(SpatialImage)
+@to_circles.register(MultiscaleSpatialImage)
+def _(
+    element: SpatialImage,
```

Comment:
Seems like you pass on `MultiscaleSpatialImage`. Mypy didn't complain?

Reply:
Fixed (didn't complain 😁)